src/Model/LocalgovAddressInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace LocalgovDrupal\OsPlacesGeocoder\Model;
+
+use Geocoder\Location;
+
+/**
+ * Adds UPRN and display value support to Location.
+ */
+interface LocalgovAddressInterface extends Location {
```

Comment:
I have added two new interfaces:
- LocationUprnInterface
- LocationOsPlacesInterface which extends LocationUprnInterface

And two new Address record classes:
- UprnAddress
- OsPlacesAddress which extends UprnAddress

Reply:
Why do they each extend each other?

To my mind if it's worth making the interfaces (and not the final classes that geocoder-php usually does) it seems useful to allow the end consumer be able to query implements UprnInterface to see if it can pull a UPRN from the location that was returned to it; implements OsGridReference added by anything that gives UK OS Grid References; equally it could query something like implements OsmId (for all the multiple geocoders that return openstreetmap way/relation/node IDs). If they are specific to the geocoder itself, then the final class suffices?

Reply:
Sorry Ekes, I have to say I haven't understood everything you have said.  I am trying to explain as best as I can :(

> If they are specific to the geocoder itself, then the final class suffices?

The OsPlacesAddress class is specific to this geocoder.  That's why I have marked it as final.

The UprnAddress class (and the LocationUprnInterface interface) are generic and can be used by any geocoder wishing to deal with UPRNs.  This geocoder is probably not the best place to house these, but then I can't think of any other place without rolling a new package.  That's why I have just left them here.

Reply:
Easier to write in code than words I guess:

https://github.com/localgovdrupal/localgov_os_places_geocoder_provider/commit/f8da8f83c8b791ef33eaa89dc0ce31292c0fedc0

To have a OS Grid Reference doesn't need to have a UPRN (and less so, but still potentially true for a geocoder vice versa). So the interfaces don't need to inherit from each other. There's a bit more explanation in my example commit message.

Reply:
> To have a OS Grid Reference doesn't need to have a UPRN

I see the disconnect now.  I was under the impression that a UPRN is an indispensable part of an Ordnance Survey Places address, so wasn't considering any Address record with just a grid reference.  Anyway, [f8da8f8](https://github.com/localgovdrupal/localgov_os_places_geocoder_provider/commit/f8da8f83c8b791ef33eaa89dc0ce31292c0fedc0) looks good.  Should we merge the [interfaces](https://github.com/localgovdrupal/localgov_os_places_geocoder_provider/tree/interfaces) branch then?